### PR TITLE
stm32f4: flash: Refactor FLASH_ACR code

### PR DESCRIPTION
**Pull request by 0xdec**
Instead of enabling data and instruction caches through flash_set_ws(), use their dedicated functions. This seemed hacky to be using flash_set_ws() to set bits other than the wait state bits. I changed it to instead use the flash_dcache_enable() and flash_icache_enable() functions, which should be more robust.

### CODE REVIEW

File: lib/stm32/f4/rcc.c [hunk 1]
```diff
@@ -709,14 +693,14 @@ uint32_t rcc_system_clock_source(void)
 
 void rcc_clock_setup_hse_3v3(const struct rcc_clock_scale *clock)
 {
-	/* Enable internal high-speed oscillator. */
+	/* Enable internal high-speed oscillator (HSI). */
```
Comment (karlp) [inline lib/stm32/f4/rcc.c]: out of scope.  please avoid making changes like this.

File: lib/stm32/f4/rcc.c [hunk 2]
```diff
@@ -743,6 +727,8 @@ void rcc_clock_setup_hse_3v3(const struct rcc_clock_scale *clock)
 	rcc_wait_for_osc_ready(RCC_PLL);
 
 	/* Configure flash settings. */
+	flash_dcache_enable();
+	flash_icache_enable();
```
Comment (karlp) [inline lib/stm32/f4/rcc.c]: hacky as it may have been, you've removed the option of running with these disabled.

File: lib/stm32/f4/rcc.c (hunk 1 above)
Comment (0xdec) [inline lib/stm32/f4/rcc.c]: Okay, I was just trying to improve the comments. Should I edit the commit to remove those, and submit a separate PR? Or just nix them entirely?

File: lib/stm32/f4/rcc.c (hunk 2 above)
Comment (0xdec) [inline lib/stm32/f4/rcc.c]: Good point. Would it be acceptable to have a check for those bits there and then call the associated flash_Xcache_enable() function if needed? That way the config options could still be passed in the struct but would not rely on flash_set_ws().

File: lib/stm32/f4/rcc.c (hunk 1 above)
Comment (karlp) [inline lib/stm32/f4/rcc.c]: I'd edit the commit to remove those, and honestly, right now, don't submit a new PR for those just now :|

File: lib/stm32/f4/rcc.c (hunk 2 above)
Comment (karlp) [inline lib/stm32/f4/rcc.c]: up to you.  You could also add bools to the struct.  You could also simply document that it's always turned on now, and see if anyone actually complains :)

File: lib/stm32/f4/rcc.c (hunk 1 above)
Comment (0xdec) [inline lib/stm32/f4/rcc.c]: Sounds good, I'm on it.

File: lib/stm32/f4/rcc.c (hunk 2 above)
Comment (0xdec) [inline lib/stm32/f4/rcc.c]: Haha, thanks!

**Comment by karlp** [comment]
this has now landed in 6798cee..90d8bd6 thanks!
